prime-router/src/main/kotlin/azure/MetaDataFunction.kt

```diff
@@ -0,0 +1,94 @@
+package gov.cdc.prime.router.azure
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.microsoft.azure.functions.ExecutionContext
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import com.microsoft.azure.functions.annotation.StorageAccount
+import org.apache.logging.log4j.kotlin.Logging
+
+/**
+ * The data class that wraps the LIVD data
+ */
+data class LivdData(
+    val manufacturer: String,
+    val model: String,
+    val vendorAnalyteName: String,
+    val vendorSpecimenDescription: List<String>,
+    val vendorResultDescription: List<String>,
+    val testPerformedLoincCode: String,
+    val testPerformedLoincLongName: String,
+    val testOrderedLoincCode: String,
+    val testOrderedLoincLongName: String,
+    val vendorComment: String,
+    val vendorAnalyteCode: String,
+    val vendorReferenceId: String,
+    val testKitNameId: String,
+    val testKitNameIdType: String,
+    val equipmentUid: String,
+    val equipmentUidType: String,
+    val component: String,
+    val property: String,
+    val time: String,
+    val system: String,
+    val scale: String,
+    val method: String,
+    val publicationVersionId: String,
+    val loincVersionId: String,
+)
+
+/**
+ * some common metadata querying functions
+ */
+class MetaDataFunction : Logging {
+    private val workflowEngine = WorkflowEngine()
+
+    /**
+     * An endpoint for getting LIVD values
+     */
+    @FunctionName("getLivdData")
+    @StorageAccount("AzureWebJobsStorage")
+    fun livdReport(
+        @HttpTrigger(
+            name = "getLivdData",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "metadata/livd"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        val filters = request.queryParameters
+        val rows = getLivdTable(filters) ?: return HttpUtilities.internalErrorResponse(request)
+        logger.info("Found ${rows.count()} rows for LIVD table.")
+        val jsonMapper = jacksonObjectMapper()
+        val responseBody = jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsString(rows)
+        // create response body
+        return HttpUtilities.okResponse(request, responseBody)
+    }
+
+    /**
+     * gets the LIVD table filtered by the values passed in
+     */
+    private fun getLivdTable(filters: Map<String, String>): List<LivdData>? {
+        logger.info("Getting the LIVD table for the metadata api")
+        val livdTable = workflowEngine
+            .metadata
+            .findLookupTable("LIVD-SARS-CoV-2-latest") ?: return null
+        logger.info("Pulling rows")
```

Comment:
Do we want to do any validation / look for jagged CSV / verify the column count of at least the header row?  Ideally the file would always be properly formatted but since this is no longer directly tied to it there may be problems in the future if a file is badly formatted.